two compiler warnings

I didn't check if these are real bugs, but they are suspicious warnings. And I'm using a recent gcc so maybe these haven't been noticed by anyone else?

```
$ gcc --version
gcc (GCC) 10.1.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fPIC -UNDEBUG -Ilib -Ilib/subprojects/tskit -Ilib/subprojects/tskit/tskit -Ilib/subprojects/kastore -I/usr/include -I/usr/include/python3.8 -I/usr/lib/python3.8/site-packages/numpy/core/include -c _msprimemodule.c -o build/temp.linux-x86_64-3.8/_msprimemodule.o -std=c99
_msprimemodule.c: In function ‘Simulator_get_nodes’:
_msprimemodule.c:242:26: warning: ‘node.metadata’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  242 |     PyObject* metadata = make_metadata(r->metadata, (Py_ssize_t) r->metadata_length);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fPIC -UNDEBUG -Ilib -Ilib/subprojects/tskit -Ilib/subprojects/tskit/tskit -Ilib/subprojects/kastore -I/usr/include -I/usr/include/python3.8 -I/usr/lib/python3.8/site-packages/numpy/core/include -c lib/interval_map.c -o build/temp.linux-x86_64-3.8/lib/interval_map.o -std=c99
lib/interval_map.c: In function ‘interval_map_alloc_single’:
lib/interval_map.c:73:31: warning: array subscript 1 is outside array bounds of ‘double[1]’ [-Warray-bounds]
   73 |         self->value[j] = value[j];
      |                          ~~~~~^~~
lib/interval_map.c:81:80: note: while referencing ‘value’
   81 | c_single(interval_map_t *self, double sequence_length, double value)
      |                                                        ~~~~~~~^~~~~

```

--- a/_msprimemodule.c
+++ b/_msprimemodule.c
@@ -4133,6 +4133,7 @@ Simulator_get_nodes(Simulator *self)
         node.time = nodes->time[j];
         node.population = nodes->population[j];
         node.individual = nodes->individual[j];
+        node.metadata = NULL;
         node.metadata_length = 0;
         py_node = make_node(&node);
         if (py_node == NULL) {

--- a/lib/interval_map.c
+++ b/lib/interval_map.c
@@ -81,7 +81,8 @@ int MSP_WARN_UNUSED
 interval_map_alloc_single(interval_map_t *self, double sequence_length, double value)
 {
     double position[2] = { 0, sequence_length };
-    return interval_map_alloc(self, 2, position, &value);
+    double value_[2] = { value, 0 };
+    return interval_map_alloc(self, 2, position, value_);
 }
 
 int

